agent_plugin_builder/build_plugin.py

```diff
@@ -0,0 +1,252 @@
+import json
+import logging
+import tarfile
+from importlib import import_module
+from pathlib import Path
+
+import yaml
+from monkeytypes import AgentPluginManifest
+
+from .build_options import PlatformDependencyPackagingMethod, parse_agent_plugin_build_options
+from .vendor_dirs import (
+    check_if_common_vendor_dir_possible,
+    generate_common_vendor_dir,
+    generate_requirements_file,
+    generate_vendor_dirs,
+)
+
+logger = logging.getLogger(__name__)
+
+
+BUILD = "build"
+DIST = "dist"
+CONFIG_SCHEMA = "config-schema.json"
+SOURCE = "source"
+MANIFEST = "manifest"
+
+
+def build_agent_plugin(
+    plugin_path: Path,
+    build_dir_path: Path,
+    dist_dir_path: Path,
+):
+    """
+    Build the agent plugin by copying the plugin code to the build directory,
+    generating the Agent Plugin archive.
+
+    :param plugin_path: Path to the plugin code.
+        If the directory does not exist, an error will be raised
```

Comment:
```suggestion
    :param plugin_path: Path to the plugin code.
```

Instead, add ":raises FileNotFoundError: If blah blah".